std/xml.d

```diff
@@ -2045,8 +2045,8 @@ class ElementParser
 
                     immutable(char)* p = startTag.tagString.ptr
                         + startTag.tagString.length;
-                    immutable(char)* q = tag_.tagString.ptr;
-                    text = decode(p[0..(q-p)], DecodeMode.LOOSE);
+                    immutable end = &tag_.tagString[0] - (&startTag.tagString[0] + startTag.tagString.length);
+                    text = decode(p[0 .. end], DecodeMode.LOOSE);
```

Comment:
It's certainly ugly, but I've had to fix uglier code before. :-)  I think it's possible, but it's probably better to split this PR and merge the simple parts first, then work on the major surgery this ugly piece of code's gonna need to get to a sane state.